hpedockerplugin/hpe/hpe_3par_common.py

```diff
@@ -464,6 +464,72 @@ def create_vlun(self, volume, host, is_snap, nsp=None, lun_id=None):
                               vlun_info['lun_id'],
                               nsp)
 
+    def create_snap_schedule(self, src_vol_name, schedName, snapPrefix,
+                             exphrs, rethrs, schedFrequency):
+        expHr = str(exphrs)
+        retHr = str(rethrs)
+
+        cmd = ['createsched']
+        createsnapstring = []
+        createsnapstring.append('"createsv ')
+        if exphrs is not None:
+            createsnapstring.append('-exp ' + expHr + 'h ')
+        if rethrs is not None:
+            createsnapstring.append('-retain ' + retHr + 'h ')
+        snap_string = ".@y@@m@@d@@H@@M@@S@"
+        dynamic_snap_name = snapPrefix + snap_string
+        createsnapstring.append(dynamic_snap_name + ' ' + src_vol_name + '"')
+
+        snapstring = ''.join(createsnapstring)
+        schedFreq = '"' + schedFrequency + '"'
+
+        cmd.append(snapstring)
+        cmd.append(schedFreq)
+        cmd.append(schedName)
+        cmd.append('\r')
+        err_resp = ""
+        try:
+            LOG.info("Creating a snapshot schedule, command is %s..." % cmd)
+            resp = self.client._run(cmd)
+            LOG.info("Created a snapshot schedule - command is: %s..." % cmd)
+            LOG.info("Create schedule response is: %s..." % resp)
+
+            for r in resp:
```

Comment:
You can refactor the code for creating a schedule/removing a schedule the CLI invocation in a common definition, so that the regex pattern detection for 'Error' string is not done twice in each operation.

Reply:
Done